projects/igniteui-angular/src/lib/combo/combo.component.ts

```diff
@@ -309,8 +316,16 @@ export class IgxComboComponent extends IgxComboBaseDirective implements AfterVie
      * ```
      */
     public select(newItems: Array<any>, clearCurrentSelection?: boolean, event?: Event) {
-        if (newItems) {
-            const newSelection = this.selectionService.add_items(this.id, newItems, clearCurrentSelection);
+        if (!newItems) {
+            return;
+        }
+
+        if (this.isRemote || (this.data && this.data.length)) {
+            const validItems = !this.isRemote
```

Comment:
We can simplify this so that we don't check if we are remote twice, something like this maybe:
```ts
let validItems = newItems;
if (!this.isRemote && this.data && this.data.length) {
    validItems = newItems.filter(item => this.isItemInData(item));
}

const newSelection = this.selectionService.add_items(this.id, validItems, clearCurrentSelection);
this.setSelection(newSelection, event);
```